src/components/pages/teacher-info-page/TeacherInformPage.jsx

```diff
@@ -21,26 +21,30 @@ const TeacherInfoPage = () => {
         const teacherData = await getUserData(teacherId, userData.token);
         setTeacherInfo(teacherData.data);
 
-        const response = await getClassesData();
-        const allClasses = response.classes;
+        try {
```

Comment:
This is a better approach. But, nesting makes this a little hard to read. Generally, you can have each action be their own function and then you can call it within the `useEffect`. That also decouples the user and class requests and allows them to work independent of each other.